pkg/operator/controller/controller.go

```diff
@@ -111,62 +110,60 @@ func (r *reconciler) reconcile(request reconcile.Request) (reconcile.Result, err
 			// This means the ingress was already deleted/finalized and there are
 			// stale queue entries (or something edge triggering from a related
 			// resource that got deleted async).
-			ingress = nil
+			logrus.Infof("clusteringress %q not found and reconciliation will be skipped", request)
 		} else {
-			return reconcile.Result{}, fmt.Errorf("failed to get clusteringress %q: %v", request, err)
+			// Try again later.
+			logrus.Errorf("failed to get clusteringress %q: %v", request, err)
+			result.RequeueAfter = 10 * time.Second
 		}
+		ingress = nil
 	}
 
 	caSecret, err := r.ensureRouterCACertificateSecret()
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to get CA secret: %v", err)
+		return reconcile.Result{}, fmt.Errorf("failed to ensure CA secret: %v", err)
 	}
 
 	// Collect errors as we go.
 	errs := []error{}
-	result := reconcile.Result{}
 
+	// Only reconcile the ingress itself if it exists.
 	if ingress != nil {
-		// Only reconcile the ingress itself if it exists.
 		dnsConfig := &configv1.DNS{}
 		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
-			errs = append(errs, fmt.Errorf("failed to get dns 'cluster': %v", err))
-			r.dnsConfig = nil
-			if errors.IsNotFound(err) {
-				result.RequeueAfter = 10 * time.Second
-			}
-		} else {
-			r.dnsConfig = dnsConfig
+			logrus.Errorf("failed to get dns 'cluster': %v", err)
+			dnsConfig = nil
 		}
-
 		infraConfig := &configv1.Infrastructure{}
 		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
-			errs = append(errs, fmt.Errorf("failed to get infrastructure 'cluster': %v", err))
-			r.infraConfig = nil
-			if errors.IsNotFound(err) {
-				result.RequeueAfter = 10 * time.Second
-			}
-		} else {
-			r.infraConfig = infraConfig
+			logrus.Errorf("failed to get infrastructure 'cluster': %v", err)
+			infraConfig = nil
 		}
-
-		// Ensure we have all the necessary scaffolding on which to place router
-		// instances.
-		err = r.ensureRouterNamespace()
-		if err != nil {
-			errs = append(errs, err)
-			return result, utilerrors.NewAggregate(errs)
-		}
-
-		if ingress.DeletionTimestamp != nil {
-			// Handle deletion.
-			if err := r.ensureIngressDeleted(ingress); err != nil {
-				errs = append(errs, fmt.Errorf("failed to ensure ingress deletion: %v", err))
-			}
+		if dnsConfig == nil || infraConfig == nil {
+			// For now, if the cluster configs are irretrievable, defer reconciliation
```

Comment:
"irretrievable" implies that we had them before and has a connotation of permanence.  Maybe "missing" or "unavailable"?

Reply:
Fixed